packages/tools/src/tools/displayTools/Contour/contourDisplay.ts

```diff
@@ -66,27 +72,91 @@ async function render(
 ): Promise<void> {
   const { segmentationId } = representationConfig;
   const segmentation = SegmentationState.getSegmentation(segmentationId);
-  const contourData = segmentation.representationData[Representations.Contour];
-  const { geometryIds } = contourData;
 
-  // We don't have a good way to handle stack viewports for contours at the moment.
-  // Plus, if we add a segmentation to one viewport, it gets added to all the viewports in the toolGroup too.
-  if (viewport instanceof StackViewport) {
+  if (!segmentation) {
     return;
   }
 
-  if (!geometryIds?.length) {
-    // This is a normal occurrence when contours activated, but not yet drawn
+  const contourData = segmentation.representationData[Representations.Contour];
+
+  const { geometryIds } = contourData;
+
+  // this means we would like to use vtk actors for contour data
+  // Note: We really should get out of
+
+  if (viewport instanceof StackViewport) {
+    // We don't have a good way to handle stack viewports for contours at the moment.
+    // Plus, if we add a segmentation to one viewport, it gets added to all the viewports in the toolGroup too.
     return;
   }
 
   // add the contour sets to the viewport
-  addOrUpdateContourSets(
+  addOrUpdateVTKContourSets(
     viewport,
     geometryIds,
     representationConfig,
     toolGroupConfig
   );
+
+  // if (contourData?.points?.length) {
```

Comment:
Do you want to leave this commented out - if so, could you add a comment as to why this is still left in the code?